docs/src/main/asciidoc/security-keycloak-authorization.adoc

```diff
@@ -559,6 +559,61 @@ quarkus.keycloak.webapp-tenant.policy-enforcer.paths.1.paths=/api/permission
 quarkus.keycloak.webapp-tenant.policy-enforcer.paths.1.claim-information-point.claims.static-claim=static-claim
 ----
 
+== Dynamic tenant configuration resolution
+
+If you need a more dynamic configuration for the different tenants you want to support and don’t want to end up
+with multiple entries in your configuration file, you can use the `io.quarkus.keycloak.pep.TenantPolicyConfigResolver`.
+
+This interface allows you to dynamically create tenant configurations at runtime:
+
+[source,java]
+----
+package org.acme.security.keycloak.authorization;
+
+import java.util.Map;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import io.quarkus.keycloak.pep.TenantPolicyConfigResolver;
+import io.quarkus.keycloak.pep.runtime.KeycloakPolicyEnforcerConfig;
+import io.quarkus.keycloak.pep.runtime.KeycloakPolicyEnforcerTenantConfig;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class CustomTenantPolicyConfigResolver implements TenantPolicyConfigResolver {
+
+    private final KeycloakPolicyEnforcerTenantConfig enhancedTenantConfig;
+    private final KeycloakPolicyEnforcerTenantConfig newTenantConfig;
+
+    public CustomTenantPolicyConfigResolver(KeycloakPolicyEnforcerConfig enforcerConfig) {
+        KeycloakPolicyEnforcerTenantConfig tenantConfig = enforcerConfig.defaultTenant();
+        this.enhancedTenantConfig = KeycloakPolicyEnforcerTenantConfig.builder(tenantConfig) <1>
```

Comment:
Builder is interesting, can we try to simplify it somehow ? There are 3 build() calls for a fairly simple policy, `get` is mentioned twice.
IMHO it is worth trying to make as compact as possible.
I'd consider dropping `add`, use concrete verb method and do `.path("enhanced-config-path").get()` or `method(GETenum)`  etc. IMHO we should have a single `build` terminating the whole sequence only.

Reply:
I will look into this. We certainly can simplify it. Single `build` I'm not sure, it would have to be `buildAll` and other parent builder aware `builds`. But things like `get()` certainly can be done. I didn't think about it as my only objection was to make it possible to configure what you can with app properties. Will try to improve it.

Reply:
I don't think we can remove what is there now as right now, it's one to one to documentation and to configuration in config properties. That makes it more verbose but clear what you need to set. I have user experience with extensive io.fabric8 k8 builders and it's hard to find right method and right path to the end.

However we can provide shortcuts for simple scenarios and document them. I'll have a look again.

Reply:
My point is, it makes sense to create "clever" methods for builders that you use often, but for builder like this one, main objection must be to have straightforward way to determine what to call.

Reply:
I added few shortucs where I though it makes sense. I don't think it makes sense to optimize this too much because you don't want to spend time studying what some method does. It's easier to have it verbous. But please feel free to suggest improvements.

Reply:
added build all option as well

Reply:
@michalvavrik Hi, the way this builder is created will serve as a blueprint for how OIDC and OIDC client configurations can be created after `new OidcTenantConfig()` and `new OidcClientConfig()` are deprecated. However, `new OidcTenantConfig()` and `new OidcClientConfig()` options are very straightforward and easy for users to use, which is why it is a quite a big deal IMHO how this builder API is designed. 

If `buildAll()` works, why have individual build() ?

Reply:
> @michalvavrik Hi, the way this builder is created will serve as a blueprint for how OIDC and OIDC client configurations can be created after `new OidcTenantConfig()` and `new OidcClientConfig()` are deprecated. 

So glad you consider this! Migration rules :-)

> However, `new OidcTenantConfig()` and `new OidcClientConfig()` options are very straightforward and easy for users to use, which is why it is a quite a big deal IMHO how this builder API is designed.

I'm in no hurry, keep asking for changes.
 
> If `buildAll()` works, why have individual `build()` ?

You have tenant config builder, and then you have nested builders. Nested builders can close all the builders with `buildAll` and build tenant config. However nested builders can also want to keep continue, like add another path, or cache or some other setting. We can't avoid nested `build`s methods or `buildAll`. Or do you have alternative proposal? Sorry, I realize you asked for something else, but I tried to explain how annoying it can be too try to figure what build method should I use. That's why this builder has also available methods that are one to one to ConfigMapping interface.

Reply:
>However nested builders can also want to keep continue, like add another path, or cache or some other setting.

IMHO this is unnecessary, I believe the focus should be on letting users do a very compact configuration creation, as opposed to keeping the flexibility like the one you described: one adds a path group, then does something else, then return to that path group to add something else. If the builder API offers an option to configure some composite set of properties, users can take that option if they need to and set whatever they need in the composite option, there is no need to keep multiple composite groups open

Reply:
That sets:

```
quarkus.keycloak.policy-enforcer.path-cache.max-entries
quarkus.keycloak.policy-enforcer.path-cache.lifespan
```

You can see that without any knowledge or thinking what  `.pathCache().maxEntries(5).lifespan(2).build()` does, I don't need to look for methods or think what they smartly do.What is alternative for this without builder?

I think this:

`.pathCache().maxEntries(5).pathCache().lifespan(2)` or maybe `pathCache().maxEntriedAndLifespan(5, 2)`

Is that what you mean? Every developer knows builder patter, I think what is there is easier, but this is hardly important for me. Just confirm @sberyozkin and there will be only one `build()` please, thank you

Reply:
Sorry, the start of previous comment disappeared :/,here it is:

 in the tests below you can see `.pathCache().maxEntries(5).lifespan(2).build()`

Reply:
Alright, I'll do that.

Reply:
@michalvavrik We can experiment with a few options, may be `set` is unnecessary as you have `pathCache()` without any parameters so it is quite distinctive.
We can look at how a builder sequence looks where all possible properties are set :-) and see what can be optimized